### PR TITLE
Improve documentation of list extensions

### PR DESCRIPTION
**Pull request by Dufgui**

### CODE REVIEW

File: devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
     @Option(shortName = 'a', hasValue = false, description = "Display name, group-id, artifact-id and version (default behaviour)")
     private boolean all = false;
 
-    @Option(shortName = 'f', hasValue = true, description = "Display in simplified format, only display the extension names or full format")
+    @Option(shortName = 'f', hasValue = true, description = "Display only extension names with name value, in simplified format with simple value or full format.")
```
Comment (cescoffier) [inline devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java]: I believe there is a typo. What does `name value` mean? 

What about: `Switch between the concise and full output. The concise output only displays the extension name`.

File: devtools/maven/src/main/java/io/quarkus/maven/ListExtensionsMojo.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
  * List the available extensions.
  * You can add one or several extensions in one go, with the 2 following mojos:
  * {@code add-extensions} and {@code add-extension}.
- * You can list all extension or just installable and choose simple or full format.
+ * You can list all extension or just installable and choose name, simple or full format.
```
Comment (cescoffier) [inline devtools/maven/src/main/java/io/quarkus/maven/ListExtensionsMojo.java]: choose between 3 output format: `name`, `simple` and `full`. (can you add a few words for each)

File: devtools/maven/src/main/java/io/quarkus/maven/ListExtensionsMojo.java [hunk 3]
```diff
@@ -34,7 +34,18 @@
...
 
     /**
-     * Display in simplified format.
+     * Display :
+     * <ul>
+     * <ul>
+     * only extension names with name value
+     * </ul>
+     * <ul>
+     * in simplified format with simple value
```
Comment (cescoffier) [inline devtools/maven/src/main/java/io/quarkus/maven/ListExtensionsMojo.java]: Shouldn't it be concise?

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java (hunk 1 above)
Comment (Dufgui) [reply devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java]: I am not very good in english, i must stop to PR dor documentation but, I just want to say there 3 format: name, simple and full. The name only display extension name, the simple add a human readable label and the full add the version detection

**Comment by Dufgui** [comment]
You can review this new version.

File: devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java [hunk 4]
```diff
@@ -23,10 +23,10 @@
     @Option(shortName = 'h', hasValue = false)
     private boolean help;
 
-    @Option(shortName = 'a', hasValue = false, description = "Display name, group-id, artifact-id and version (default behaviour)")
+    @Option(shortName = 'a', hasValue = false, description = "Display all extensions or just the installable.")
```
Comment (cescoffier) [inline devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java]: what about _compatible_ ones?

File: devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java [hunk 5]
```diff
@@ -23,10 +23,10 @@
...
     private boolean all = false;
 
-    @Option(shortName = 'f', hasValue = true, description = "Display in simplified format, only display the extension names or full format")
+    @Option(shortName = 'f', hasValue = true, description = "Choose between 3 output formats: name, simple, full.")
```
Comment (cescoffier) [inline devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java]: What about `Select the output format among 'name' (display the name only), 'simple' (display ...) and 'full' (default).`

I also wondering if `simple` should not be replaced by `concise`.

File: devtools/maven/src/main/java/io/quarkus/maven/ListExtensionsMojo.java [hunk 6]
```diff
@@ -16,7 +16,7 @@
  * List the available extensions.
  * You can add one or several extensions in one go, with the 2 following mojos:
  * {@code add-extensions} and {@code add-extension}.
- * You can list all extension or just installable and choose simple or full format.
+ * You can list all extension or just installable. Choose between 3 output formats: name, simple and full.
```
Comment (cescoffier) [inline devtools/maven/src/main/java/io/quarkus/maven/ListExtensionsMojo.java]: Same comment as above.

**Review by cescoffier — CHANGES_REQUESTED** [review]
I've made another review. Still some comments.

What about renaming `simple` to `concise`. I'm not a big fan of _simple_.

File: devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java (hunk 4 above)
Comment (Dufgui) [reply devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java]: In fact we have no notion of compatibility, we just check if this already installed or not

Comment (Dufgui) [reply devtools/aesh/src/main/java/io/quarkus/cli/commands/ListExtensionsCommand.java]: In fact we have no notion of compatibility, we just check if this already installed or not

**Comment by Dufgui** [comment]
it strange, the commit is not sync between the PR and my branch

**Review by cescoffier — APPROVED** [review]

**Pull request merged**